Dynamic Job Segmentation for UF

VasanthSadhasivan opened the pull request.

VasanthSadhasivan committed:
feat: TER-78 Setup docker based segmentation tooling

This commit introduces support to push jobs onto a separate
docker container managing jobs. This docker container itself
runs docker for each job that will be running.

Right now, we assume that jobs will exit after they are done
being used by spectralis. They act as a kind of of one-shot.

jdrakes commented on backend/adumbra/workers/tasks/image_segment.py:
```diff
@@ -0,0 +1,89 @@
+            "input_schema": """{"images":[{"blob":"", "id":""}]}""",
+        }
+        response = requests.post(
+            "http://jobs:6543/api/job_infos", files=files, data=data, timeout=120
```
jobs:6543 can be interpolated into the request strings in this file

jdrakes commented on docker-compose.yml:
```diff
@@ -66,6 +68,28 @@ services:
+      adumbra_jobs_network:
+        aliases:
+          - jobs
+    privileged: true
```
What does this do?

VasanthSadhasivan replied:
So the new jobs service was to create a new docker sub container which will run the jobs api as well as host docker in docker containers running the passed in, UF custom algorithms.

The jobs_docker_data volume I believe was added to persist docker volumes, containers and images. I think I added this initially when i wanted to do more fancy things with the job manager and with docekr, but right now, I am naively creating docker containers each time a user uploads a docker image. i didnt spend time cleaning that up for spectralis, but I think that is valid clean up task for drawinsight jobs, ill add a ticket (TER-48).

I think that the ports line can be removed since the docker containers that need to talk to the jobs api are running in the same `adumbra_jobs_network` docker network.

`adumbra_jobs_network` is a network which allowed for containers to talk to each other (namely, the celery worker to send post requests to this new jobs container).

the `privileged ` and `environment` fields were _I believe_ added to get docker installed inside of this jobs container. I can try to remove some of these and see if things still work? But I don't remember exactly why they were needed :/

VasanthSadhasivan committed:
feat: TER-78 Update the job manager API plus few docker setup changes

In this commit, we are updating hte job manager API to expose an endpoint
to vew all jobs available on the jobs docker container. We update the
Vue code to show existing jobs to select from or upload a new job.

We have to add the frontend to make a request to the jobs api directly,
so we update the docker file to add the frontend container to attach to
the jobs docker network, and we also add a proxy route to the route api
endpoint.

We also expose the port 8080 on the frontend docker container.

VasanthSadhasivan committed:
noop: formatting fix post running black

VasanthSadhasivan committed:
noop: allow for xrayrecon package to use python 3.12.10
